File: IncludeDataBinding/app/src/main/java/com/androidstudiomanju/includedatabinding/MainActivity.java
package com.androidstudiomanju.includedatabinding;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;

import android.os.Bundle;

import com.androidstudiomanju.includedatabinding.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {
    private ActivityMainBinding activityMainBinding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Person person1 = new Person("Jack");
        activityMainBinding = DataBindingUtil.setContentView(this,R.layout.activity_main);
        activityMainBinding.setPerson(person1);

    }
}

File: AndroidKotlin/FragmentsApp/app/src/main/java/com/androidstudiomanju/fragmentsapp/MainActivity.kt
package com.androidstudiomanju.fragmentsapp

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentTransaction
import com.androidstudiomanju.Fragment1

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val btn1:Button = findViewById(R.id.btn1)
        val btn2:Button = findViewById(R.id.btn2)
        btn1.setOnClickListener(){
            val fragment1:Fragment = Fragment1()
            val fragment2:Fragment = Fragment2()
            val frtrans:FragmentTransaction = supportFragmentManager.beginTransaction()
            frtrans.replace(R.id.frame1,fragment1).commit()
        }
        btn2.setOnClickListener(){
            val fragment2:Fragment = Fragment2()
            val frtrans:FragmentTransaction = supportFragmentManager.beginTransaction()
            // normal Program use the below code
            //frtrans.replace(R.id.frame2,fragment2).commit()
            // testing the lifecycle of fragment replacing fragment1 with fragment2
            frtrans.replace(R.id.frame1,fragment2).commit()
        }
    }
}